crates/starknet/src/starknet/add_declare_transaction.rs

```diff
@@ -34,6 +40,9 @@ pub fn add_declare_transaction_v3(
         .execute(&mut starknet.state.state, &starknet.block_context, true, true);
 
     starknet.handle_transaction_result(transaction, blockifier_execution_result)?;
+    starknet.handle_dump_event(DumpEvent::AddDeclareTransaction(
```

Comment:
In `handle_transaction_result()` we have transaction as`Transaction` type not `BroadcastedTransaction` 

I tried to avoid that kind of refactor but yes it's doable, should I do it?

Reply:
Discussed on a call. Conclusion: leave as is. Resolving.